frontend/packages/operator-lifecycle-manager/integration-tests-cypress/tests/deprecated-operator-warnings.cy.ts

```diff
@@ -0,0 +1,227 @@
+const DEPRECATED_OPERATOR_WARNING_BADGE_ID = 'deprecated-operator-warning-badge';
+const DEPRECATED_OPERATOR_WARNING_PACKAGE_ID = 'deprecated-operator-warning-package';
+const DEPRECATED_OPERATOR_WARNING_CHANNEL_ID = 'deprecated-operator-warning-channel';
+const DEPRECATED_OPERATOR_WARNING_VERSION_ID = 'deprecated-operator-warning-version';
+
+describe('Deprecated operator warnings', () => {
+  before(() => {
+    cy.login();
+    create(testDeprecatedCatalogSource);
+  });
+
+  after(() => {
+    cy.visit('/');
+    cy.exec(
+      `oc delete subscription ${testDeprecatedSubscription.metadata.name} -n ${testDeprecatedSubscription.metadata.namespace}`,
+    );
+    cy.exec(
+      `oc delete clusterserviceversion ${testDeprecatedSubscription.spec.startingCSV} -n ${testDeprecatedSubscription.metadata.namespace}`,
+    );
+    cy.exec(
+      `oc delete ${testDeprecatedCatalogSource.kind} ${testDeprecatedCatalogSource.metadata.name} -n ${testDeprecatedCatalogSource.metadata.namespace}`,
+    );
+    checkErrors();
+  });
+
+  it('verify deprecated operator warning badge on the OperatorHub tile', () => {
+    cy.visit(
+      `/k8s/ns/${testDeprecatedCatalogSource.metadata.namespace}/operators.coreos.com~v1alpha1~CatalogSource/test-community-operator-deprecation`,
+    );
+    cy.log('verify the test-community-operator-deprecation CatalogSource is in "READY" status');
+    cy.byTestSelector('details-item-value__Status', TIMEOUT).should('have.text', 'READY');
+
+    cy.log('visit OperatorHub');
+    cy.visit('/operatorhub/all-namespaces');
+
+    cy.log('filter by the group name');
+    cy.byTestID('source-community-operators-for-testing-deprecation').click();
+
+    cy.log('filter by the operator name');
+    cy.byTestID('search-operatorhub').type(testOperatorName);
+    cy.get('.co-catalog-tile', TIMEOUT).its('length').should('eq', 1);
+
+    cy.log('verify the Deprecated badge on Kiali Community Operator tile');
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_BADGE_ID)
+      .contains(deprecatedBadge)
+      .should('be.visible');
+  });
+
+  it('verify deprecated operator warnings on the Install Operator modal', () => {
+    cy.log('visit Install Operator modal');
+    cy.visit(
+      '/operatorhub/all-namespaces?keyword=kia&details-item=kiali-test-community-operator-deprecation-openshift-marketplace&channel=stable&version=1.83.0',
+    );
+    cy.log('verify the Deprecated badge on Kiali Community Operator logo');
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_BADGE_ID)
+      .contains(deprecatedBadge)
+      .should('be.visible');
+
+    cy.log('verify the package deprecation warning exists');
+    operator.modalElementShouldBeVisible(
+      DEPRECATED_OPERATOR_WARNING_PACKAGE_ID,
+      deprecatedPackageMessage,
+    );
+    cy.byTestID(DEPRECATED_OPERATOR_WARNING_CHANNEL_ID)
+      .contains(deprecatedChannelMessage)
+      .should('not.exist');
```

Comment:
I would have to validate if this is functioning as expected since there is hidden CSS in the modal.

Reply:
> I would have to validate if this is functioning as expected since there is hidden CSS in the modal.

Never mind. This is working as expected.